commit 3f9737e718367c083be3b35d2258b2f241c43dc6
Author: Ryan Burns <rtburns@jpl.nasa.gov>
Date:   Mon Oct 2 13:45:32 2023 -0700

    Add missing annotations import
    
    Needed for python 3.8 support

diff --git a/python/packages/nisar/mixed_mode/logic.py b/python/packages/nisar/mixed_mode/logic.py
index d5f729504..37e354aa1 100644
--- a/python/packages/nisar/mixed_mode/logic.py
+++ b/python/packages/nisar/mixed_mode/logic.py
@@ -1,3 +1,4 @@
+from __future__ import annotations
 from collections import defaultdict
 from dataclasses import dataclass
 from nisar.products.readers.Raw.Raw import RawBase
