diff --git a/python/packages/isce3/signal/rfi_freq_null.py b/python/packages/isce3/signal/rfi_freq_null.py
index e02e19c7e..e8c76fdc7 100644
--- a/python/packages/isce3/signal/rfi_freq_null.py
+++ b/python/packages/isce3/signal/rfi_freq_null.py
@@ -3,6 +3,8 @@ Perform RFI Detection and mitigation by evaluating raw data frequency domain
 averaged Power Spectra. Detected RFI-contaminated frequency bins are nulled.
 """
 
+from __future__ import annotations
+
 import numpy as np
 from scipy.fft import fft, ifft
 from numpy.polynomial import Polynomial
diff --git a/python/packages/nisar/products/readers/antenna/antenna_parser.py b/python/packages/nisar/products/readers/antenna/antenna_parser.py
index 58ca9b1f7..1d14f1ffa 100644
--- a/python/packages/nisar/products/readers/antenna/antenna_parser.py
+++ b/python/packages/nisar/products/readers/antenna/antenna_parser.py
@@ -1,3 +1,5 @@
+from __future__ import annotations
+
 import h5py
 import numpy as np
 import re
